### DISCUSSION TITLE
ImportError: DLL load failed while importing ie_api: The specified module could not be found.

### DISCUSSION THREAD

**Discussion by meng032020** [openvinotoolkit/openvino_notebooks#40] [Q&A] [answered]
Hi, I encountered the below error, when I run the 001-hello-world. 
Do you know how to solve it? Do I need to install OpenVINO first before going through this Installation Guide? Thanks!

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-c40e477b029a> in <module>
      4 import matplotlib.pyplot as plt
      5 import numpy as np
----> 6 from openvino.inference_engine import IECore

~\openvino_notebooks\openvino_env\lib\site-packages\openvino\inference_engine\__init__.py in <module>
     32         os.environ["PATH"] = os.path.abspath(openvino_dlls) + ";" + os.environ["PATH"]
     33 
---> 34 from .ie_api import *
     35 __all__ = ['IENetwork', "TensorDesc", "IECore", "Blob", "PreProcessInfo", "get_version"]
     36 __version__ = get_version()

ImportError: DLL load failed while importing ie_api: The specified module could not be found.

**Comment by helena-intel** [comment]
Hi! The README covers all the steps, you do not need to install OpenVINO through other channels. `pip install -r requirements.txt` installs the OpenVINO development package and all other required Python packages. There is a `check_install.py` script in the root of the repository. Could you run `python check_install.py` and post the output?

**Reply by meng032020** [reply]
Hi Helena,

Thank you for the prompt reply.
The output as below:
Thank you for your help!

![image](https://user-images.githubusercontent.com/69293028/116020067-23b0ae80-a678-11eb-873e-005a7ae0cd05.png)

**Comment by helena-intel** [comment, marked as answer]
Thanks @meng032020 , that is helpful! I see that your Python is from anaconda/miniconda (because of (base)  after (openvino_env)). It is very likely that that causes the error. I will test this and write instructions for this situation. In the meantime, if you want to try the notebooks, you could install Python from https://www.python.org/ftp/python/3.7.9/python-3.7.9-amd64.exe . If you are using Anaconda for other projects, you should not select the "add Python to your Windows PATH" box. If you install it for your current user, the executable will then be `C:\Users\<username>\AppData\Local\Programs\Python\Python37\python.exe` and you open a new command prompt, and make the virtualenv by typing (replace <username> with your username)

```
C:\Users\<username>\AppData\Local\Programs\Python\Python37\python.exe -m venv openvino_env
```

and then follow the instructions starting with step 3. To be safe, I would remove the existing Jupyter kernel for openvino_env with `jupyter kernelspec remove openvino_env` before installing it again (in step 5). 

This is just a temporary workaround though, the notebooks should work with your conda python too, and I will look into that.

**Reply by meng032020** [reply]
Hi Helena,
Thank you so much for the pointers and apologies for the late reply.
I tried the steps that you mentioned, to download the Python and run the executable.
However, the executable is directly inside python codes and not like a command line, how can I type the command?

![image](https://user-images.githubusercontent.com/69293028/116770013-01050800-aa73-11eb-99c2-9d7da8d00856.png)

![image](https://user-images.githubusercontent.com/69293028/116770023-18dc8c00-aa73-11eb-91c2-1def517f0ab1.png)

**Reply by meng032020** [reply]
Hi Helena, 
Please ignore the above message, found out that the actual method is just to type your command using any command prompt =)
However, I encountered a problem with Step 5, could you kindly help to advise? Thanks!

![image](https://user-images.githubusercontent.com/69293028/116770290-327ed300-aa75-11eb-87b4-82456730bb05.png)

**Reply by helena-intel** [reply]
The "no space left on device" error means that your c:\ drive is full. The notebook requirements install some large dependencies like TensorFlow and PyTorch. At least 2GB free space is necessary, possibly a bit more during installation. If you deleted a previous openvino_env folder it may still be in your Recycle Bin and still take up space. I hope this was the last hurdle!

**Reply by meng032020** [reply]
Hi Helena, yeah, my laptop was running low in space. OK, so what I did is I uninstall everything and reinstalled everything again.

I don't really remember the exact steps because I was following the actual guide and trying your suggestions, but I managed to get the notebook running. =) 

However, the below observation is quite strange (attached photo). Check_accuracy.py works for python (I assume this is from anaconda) but failed on the python38. Do you know if this is expected?
And would this work on Python3.8 or only Python3.7?

And for some reasons, I would need to install only Anaconda only (without the python as standalone), hence would be waiting for the testing on this too. Appreciate your help for all this time. Thank you=)

![image](https://user-images.githubusercontent.com/69293028/116841627-51fc3400-ac0c-11eb-8aac-be25c5deca23.png)
